tensilelite/Tensile/KernelWriter.py

```diff
@@ -3967,11 +3981,19 @@ def readWriteVectors(mat, vw, kernel):
     if not (kernel["ProblemType"]["Sparse"] and not kernel["DirectToVgprSparseMetadata"]):
       self.states.m.numVgprLocalWriteAddr = 0
     # do not allocate local write address register if DirectToVgpr is enabled
-    if kernel["DirectToVgprA"]:
+    if kernel["DirectToVgprA"] or kernel["DirectToLdsA"]:
       self.states.a.numVgprLocalWriteAddr = 0
-    if kernel["DirectToVgprB"]:
+    if kernel["DirectToVgprB"] or kernel["DirectToLdsB"]:
       self.states.b.numVgprLocalWriteAddr = 0
 
+    if kernel["StoreSwapAddr"]:
+      self.states.a.numVgprLocalReadSwapAddr = 1
```

Comment:
Why this is always 1?
Should this be same as numVgprLocalReadAddr?

Reply:
My understanding is that we only have multiple lra/lwa to handle cases where the instruction offsets (LDS) would be over 64KB - I'm not sure if there are other use cases. For `StoreSwapAddr` the instruction offset will be less than 64KB since we move the offsets to the vgprs when trying to swap between lds buffers.

Reply:
If we do not need multiple vgprs for local read/write address for StoreswapAddr, we need to disable allocating multiple vgprs for local read/write if StoreswapAddr is true.